### PR TITLE
Add volume controller

### LINKED ISSUE #56: Volume controller?
Would be nice to have a volume controller build into player itself. When there are couple of audio streams (from different sources), that would be useful.

### PR DESCRIPTION
**Pull request by v12**
Volume controller that handles both clicks and wheel (trackpad) movement when mouse is hovering the controller 😎

Closes #56

### CODE REVIEW

**Comment by MichielDeMey** [comment]
Nice PR, but I would like to see us use and skin the native range input element, that will allow us to read the value from the native element.

**Comment by v12** [comment]
@MichielDeMey fair point, I absolutely forgot that input[type=range] exists 👍

File: app/js/components/mediaPlayer.js [hunk 1]
```diff
@@ -112,8 +114,53 @@ var MediaPlayer = React.createClass({
...
+
+    // make sure new volume level is in range [0,1]
+    if (newVolume > 1)
+      newVolume = 1
+    else if (newVolume < 0) {
+      newVolume = 0
+    }
+
+    this.setState({
+      lastVolume: volume,
```
Comment (MichielDeMey) [inline app/js/components/mediaPlayer.js]: Let's make sure we have a fixed length.
Using `volume.toFixed(2)` will truncate the volume level to 2 numbers at the most.

e.g. `0.45555555` will become `0.45`

**Comment by MichielDeMey** [comment]
Let's also persist the volume level (but not the "muted" state) to localStorage when the user closes the application, I believe that would improve UX a bit.

**Comment by v12** [comment]
@MichielDeMey just pushed changes that implement all your suggestions 👍

**Comment by v12** [comment]
Just wondering if anything prevents this PR from being merged? ☺️

**Comment by MichielDeMey** [comment]
Whoops, we totally lost track of this issue.

Thank you for your contribution!

**Comment by fabianperez** [comment]
Just came here looking for volume control support. Thanks for doing this! 🙌

**Comment by gillesdemey** [comment]
Released as [v0.8.0](https://github.com/gillesdemey/Cumulus/releases/tag/v0.8.0)

**Comment by FoxxMD** [comment]
Thanks for all the hard work!!

**Pull request merged**